src/providers/pdal/qgspdaleptgenerationtask.cpp

```diff
@@ -0,0 +1,181 @@
+/***************************************************************************
+  qgspdaleptgenerationtask.cpp
+  ------------------------
+  Date                 : December 2020
+  Copyright            : (C) 2020 by Peter Petrik
+  Email                : zilolv at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgspdaleptgenerationtask.h"
+
+#include <vector>
+#include <string>
+#include <QDebug>
+#include <QThread>
+#include <QFileInfo>
+#include <QDir>
+#include <QProcessEnvironment>
+
+#include "qgsapplication.h"
+#include "QgisUntwine.hpp"
+#include "qgsmessagelog.h"
+#include "qgis.h"
+
+QgsPdalEptGenerationTask::QgsPdalEptGenerationTask( const QString &file, const QString &outputDir )
+  : QgsTask( tr( "Generate EPT Index" ) )
+  , mOutputDir( outputDir )
+  , mFile( file )
+{
+  mUntwineExecutableBinary = guessUntwineExecutableBinary();
+}
+
+bool QgsPdalEptGenerationTask::run()
+{
+  if ( isCanceled() || !prepareOutputDir() )
+    return false;
+
+  if ( isCanceled() || !runUntwine() )
+    return false;
+
+  if ( isCanceled() )
+    return false;
+
+  cleanTemp();
+
+  return true;
+}
+
+void QgsPdalEptGenerationTask::cleanTemp()
+{
+  QDir tmpDir( mOutputDir + QStringLiteral( "/temp" ) );
+  if ( tmpDir.exists() )
+  {
+    QgsMessageLog::logMessage( tr( "Removing temporary files in %1" ).arg( tmpDir.dirName() ), QObject::tr( "Untwine" ), Qgis::Info );
```

Comment:
Instead of using `QObject::tr( "Untwine" )` for the category I would use non-translatable string (it's a project name) or maybe even better, translatable string e.g. "Point clouds"